api/resources/component.go

```diff
@@ -55,19 +58,53 @@ type Component struct {
 	Name                   string                 `json:"name"`
 }
 
+type CPUQuantity struct {
+	resource.Quantity
+}
+
+func (c *CPUQuantity) MarshalJSON() ([]byte, error) {
+	capInStr := strconv.FormatInt(c.MilliValue(), 10)
+	return []byte(fmt.Sprintf(`"%s"`, capInStr)), nil
+}
+
+type MemoryQuantity struct {
+	resource.Quantity
+}
+
+func (m *MemoryQuantity) MarshalJSON() ([]byte, error) {
+	capInStr := strconv.FormatInt(m.Value(), 10)
+	return []byte(fmt.Sprintf(`"%s"`, capInStr)), nil
+}
+
 type ComponentDetails struct {
-	*Component           `json:",inline"`
+	Name string `json:"name"`
+
+	v1alpha1.ComponentSpec `json:",inline"`
+
+	// hack to override & ignore field in ComponentSpec
+	//ResourceRequirements interface{} `json:"resourceRequirements,omitempty"`
+
+	CPURequest    *CPUQuantity    `json:"cpuRequest,omitempty"`
```

Comment:
Why not declare the following 4 feilds in Component struct?

Reply:
- originally, I put the fields Component, then I found several levels of embed struct in ComponentDetails (ComponentDetails embed Component which embeds ComponentSpec) makes the code confusing in some cases:
  
![image](https://user-images.githubusercontent.com/48272667/88875289-9ad52580-d253-11ea-9c84-c9d410fe9cbe.png)

- and it can be more clear if Component is for request and ComponentDetails is for response, these four fields only show in response, so I put them in ComponentDetails instead of Component.